apps/server/src/modules/tool/uc/lti11.uc.spec.ts

```diff
@@ -0,0 +1,278 @@
+				ltiRoleMapper.mapRoleToLtiRole.mockReturnValue(LtiRole.LEARNER);
+				userService.getUser.mockResolvedValue(userDto);
+				lti11Service.getUserId.mockResolvedValue(currentUser.userId);
+				userService.getDisplayName.mockResolvedValue(userName);
+				lti11Service.createConsumer.mockReturnValue(oauth);
+				oauth.authorize.mockReturnValue(authorization);
+
+				return {
+					userName,
+					currentUser,
+					userDto,
+					ltiTool,
+					authorization,
+				};
+			};
+
+			it('should use tool with resource_link_id', async () => {
+				const { currentUser, ltiTool, authorization } = setup();
+				const expectedRequestData: RequestOptions = {
+					url: ltiTool.url,
+					method: 'POST',
+					data: {
+						lti_version: ltiTool.lti_version,
+						lti_message_type: ltiTool.lti_message_type,
+						resource_link_id: ltiTool.resource_link_id,
+						roles: LtiRole.LEARNER,
+						launch_presentation_document_target: 'window',
+						launch_presentation_locale: 'en',
+						user_id: currentUser.userId,
+					},
+				};
+
+				const result: Authorization = await useCase.getLaunchParameters(currentUser, 'toolId', 'courseId');
+
+				expect(result).toEqual(authorization);
+				expect(oauth.authorize).toHaveBeenCalledWith(expect.objectContaining(expectedRequestData));
+			});
+
+			it('should use courseId as resource_link_id, if tool has none', async () => {
+				const { currentUser, ltiTool, authorization } = setup();
+				const courseId = 'courseId';
+				ltiTool.resource_link_id = undefined;
+				const expectedRequestData: RequestOptions = {
+					url: ltiTool.url,
+					method: 'POST',
+					data: {
+						lti_version: ltiTool.lti_version,
+						lti_message_type: ltiTool.lti_message_type,
+						resource_link_id: courseId,
+						roles: LtiRole.LEARNER,
+						launch_presentation_document_target: 'window',
+						launch_presentation_locale: 'en',
+						user_id: currentUser.userId,
+					},
+				};
+
+				const result: Authorization = await useCase.getLaunchParameters(currentUser, 'toolId', courseId);
+
+				expect(result).toEqual(authorization);
+				expect(oauth.authorize).toHaveBeenCalledWith(expect.objectContaining(expectedRequestData));
+			});
+
+			it('should use tool with pseudonymous privacy_permission', async () => {
+				const { currentUser, ltiTool, authorization } = setup();
+				const expectedRequestData: RequestOptions = {
+					url: ltiTool.url,
+					method: 'POST',
+					data: {
+						lti_version: ltiTool.lti_version,
+						lti_message_type: ltiTool.lti_message_type,
+						resource_link_id: ltiTool.resource_link_id,
+						roles: LtiRole.LEARNER,
+						launch_presentation_document_target: 'window',
+						launch_presentation_locale: 'en',
+						user_id: currentUser.userId,
+					},
+				};
+
+				const result: Authorization = await useCase.getLaunchParameters(currentUser, 'toolId', 'courseId');
+
+				expect(result).toEqual(authorization);
+				expect(oauth.authorize).toHaveBeenCalledWith(expectedRequestData);
```

Comment:
What is the difference between this test and the test before expect the objectContaining? Maybe you use object containing because there is more data in the result object?